### PR TITLE
[docs]grpc datasource

### PR DESCRIPTION
**Pull request by withshubh**
Documentation #6211

### CODE REVIEW

File: docs/docs/data-sources/grpc.md [hunk 1]
```diff
@@ -1,11 +1,91 @@
...
+id: grpc
+title: gRPC
+---
 
-# grpc
+<div className='badge badge--primary heading-badge'>Self-hosted only</div>
 
-ToolJet can connect to GRPC databases to read and write data. 
+:::caution
+At present, gRPC can be deployed for self-hosting and has the capability to handle unary requests and responses.
```
Comment (arpitnath) [inline docs/docs/data-sources/grpc.md:9]: Can we write something like this
```suggestion
only self-hosted deployments will have access to a gRPC datasource that is capable of handling unary requests and responses.
```

File: docs/docs/data-sources/grpc.md [hunk 2]
```diff
@@ -1,11 +1,91 @@
...
 
-## Querying GRPC
\ No newline at end of file
+### Step 1: Upgrade ToolJet to the version 2.5 or above
+
+Find instructions on how to do this in the setup guides located here: https://docs.tooljet.com/docs/setup/.
+
+### Step 2: Add proto files
+
+Add proto files by creating a directory named "protos" and placing a file called "**service.proto**" inside it.
```
Comment (arpitnath) [inline docs/docs/data-sources/grpc.md:23]: we should mention the **protos** dir should be in the root

File: docs/docs/data-sources/grpc.md [hunk 3]
```diff
@@ -1,11 +1,91 @@
...
+
+:::info Note
+Copy-pase the current service.proto service
```
Comment (arpitnath) [inline docs/docs/data-sources/grpc.md:26]: We can remove this, not relevant

File: docs/docs/data-sources/grpc.md [hunk 4]
```diff
@@ -1,11 +1,91 @@
...
+docker-compose up -d
+```
+
+## Querying gRPC
+
+After setting up your proto files, you should be able to establish a connection to gRPC by going to the [global datasource](/docs/data-sources/overview) page.
+
+### Connect the gRPC datasource
+
+ToolJet requires the following to connect to Amazon SES:
```
Comment (arpitnath) [inline docs/docs/data-sources/grpc.md:64]: gRPC servers

File: docs/docs/data-sources/grpc.md [hunk 5]
```diff
@@ -1,11 +1,91 @@
...
+
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/datasource-reference/grpc/grpcgds.png" alt="gRPC: connection" />
+
+</div>
+
+### Creating query
+
+You can now query a particular of RPC of the added services.
```
Comment (arpitnath) [inline docs/docs/data-sources/grpc.md:85]:
```suggestion
You can now query a particular of RPC method of the added services.
```

**Review by arpitnath — CHANGES_REQUESTED** [review]

File: docs/docs/data-sources/grpc.md [hunk 6]
```diff
@@ -1,11 +1,91 @@
...
 
-## Querying GRPC
\ No newline at end of file
+### Step 1: Upgrade ToolJet to the version 2.5 or above
+
+Find instructions on how to do this in the setup guides located here: https://docs.tooljet.com/docs/setup/.
+
+### Step 2: Add proto files
+
+Add proto files by creating a directory named "protos" and placing a file called "**service.proto**" inside it.
```
Comment (withshubh) [inline docs/docs/data-sources/grpc.md:23]:
```suggestion
At the root, create a directory named "**protos**" and add a "**service.proto**" file inside it.
```

File: docs/docs/data-sources/grpc.md [hunk 7]
```diff
@@ -1,11 +1,91 @@
...
+
+:::info Note
+Copy-pase the current service.proto service
+:::
```
Comment (withshubh) [inline docs/docs/data-sources/grpc.md:27]:
```suggestion
```

File: docs/docs/data-sources/grpc.md [hunk 8]
```diff
@@ -1,11 +1,91 @@
...
+docker-compose up -d
+```
+
+## Querying gRPC
+
+After setting up your proto files, you should be able to establish a connection to gRPC by going to the [global datasource](/docs/data-sources/overview) page.
+
+### Connect the gRPC datasource
+
+ToolJet requires the following to connect to Amazon SES:
```
Comment (withshubh) [inline docs/docs/data-sources/grpc.md:64]:
```suggestion
ToolJet requires the following to connect to gRPC servers:
```

File: docs/docs/data-sources/grpc.md [hunk 9]
```diff
@@ -1,11 +1,91 @@
...
+
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/datasource-reference/grpc/grpcgds.png" alt="gRPC: connection" />
+
+</div>
+
+### Creating query
+
+You can now query a particular of RPC of the added services.
```
Comment (withshubh) [inline docs/docs/data-sources/grpc.md:85]:
```suggestion
You can now query a particular RPC method of the added services.
```

**Pull request merged**